src/main/java/com/vejo/em/casa/be/rest/CategoryController.java

```diff
@@ -15,18 +21,17 @@
     @Autowired
     CategoryService service;
 
-    @GetMapping(value="getAllCategories")
+    @GetMapping()
     public List<Category> getAllCategories(){  return service.findAll(); }
 
-    @GetMapping(value = "getSingleCategory/{id}")
-    public Category getSingleCategory(@PathVariable Optional<Integer> id) {
-        if (id.isPresent()) {
-            return service.findById(id.get().longValue());
-        } else {
-            return null;
-        }
+
+    @GetMapping("{id}")
+    public Category getSingleCategory(@PathVariable Long id) {
+        return service.getSingleCategory(id);
     }
 
-    @PostMapping
-    public void save(Category category) { this.service.save(category); }
+
+    @PostMapping()
+    public Category save(Category category) { return this.service.save(category); }

```

Comment:
for now we can't have this available because the API is pubic. As such anyone can call it which means that anyone can enter data in our DB, which is not expected. We can only use it when we have it secured by a login